crates/viewer/re_space_view_graph/src/ui/draw.rs

```diff
@@ -281,29 +283,58 @@ pub fn draw_graph(
-                let response = draw_node(ui, center, node.label(), highlight);
-
-                let response = if let Label::Text { text, .. } = &instance.label {
-                    response.on_hover_text(format!(
-                        "Graph Node: {}\nLabel: {text}",
-                        instance.graph_node.as_str(),
-                    ))
-                } else {
-                    response.on_hover_text(format!("Graph Node: {}", instance.graph_node.as_str(),))
-                };
+                let mut response = draw_node(ui, center, node.label(), highlight);
 
                 let instance_path =
                     InstancePath::instance(entity_path.clone(), instance.instance_index);
                 ctx.select_hovered_on_click(
                     &response,
-                    vec![(Item::DataResult(query.space_view_id, instance_path), None)].into_iter(),
+                    vec![(
+                        Item::DataResult(query.space_view_id, instance_path.clone()),
+                        None,
+                    )]
+                    .into_iter(),
                 );
 
+                response = response.on_hover_ui_at_pointer(|ui| {
+                    list_item::list_item_scope(ui, "graph_node_hover", |ui| {
+                        item_ui::instance_path_button(
+                            ctx,
+                            &query.latest_at_query(),
+                            ctx.recording(),
+                            ui,
+                            Some(query.space_view_id),
+                            &instance_path,
+                        );
+
+                        instance_path.data_ui_recording(ctx, ui, UiLayout::Tooltip);
+                    });
+                });
+
                 response
             }
-            Node::Implicit { graph_node, .. } => {
-                draw_node(ui, center, node.label(), Default::default())
-                    .on_hover_text(format!("Implicit Graph Node: {}", graph_node.as_str(),))
+            Node::Implicit { edge_instance, .. } => {
+                let mut response = draw_node(ui, center, node.label(), Default::default());
+
+                // TODO(#6889): @grtlr This is only somewhat correct until we have tagged components.
+                let instance_path = InstancePath::instance(entity_path.clone(), *edge_instance);
+
+                response = response.on_hover_ui_at_pointer(|ui| {
+                    list_item::list_item_scope(ui, "graph_edge_hover", |ui| {
+                        item_ui::instance_path_button(
+                            ctx,
+                            &query.latest_at_query(),
+                            ctx.recording(),
+                            ui,
+                            Some(query.space_view_id),
+                            &instance_path,
+                        );
+
+                        instance_path.data_ui_recording(ctx, ui, UiLayout::Tooltip);
```

Comment:
Because of my comment above and because this is generally somewhat confusing (and soon broken), I would suggest not having this here, and instead just a text stating "Implicit node created via a reference in the `GraphEdge` component" or something of that nature.

The edge instance tooltip should be displayed when the actual edge is hovered imo, but that can be left for later (btw, kurbo has point to bezier distance computation so this is something it could help with).

Reply:
That makes sense, and I agree. I will change it back!